### ISSUE TITLE
Create an index on identity_provider_links(users_id) to speed up queries

### ISSUE THREAD

**Issue by andrewpai** [FusionAuth/fusionauth-issues#2122]
### Description
There is a query against identity_provider_links that includes a user id in the where clause, which should typically yield a small number of records in the query result. Currently, a query for a single user id can result in scanning a large number of rows. Adding an index to this column should greatly improve the performance of this query.

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

**Comment by kchenery** [comment]
This is an issue that has affected us.  The api call that generates the query is: `GET /api/identity-provider/link?identityProviderId={identityProviderId}&userId={userId}`

The unique index on the table can only be used for part of the `WHERE` clause in the resulting query.  It is only the first key column that can be considered and we have instances where this matches many thousands of rows (hundreds of thousands in one case) which results in the large scan.

When we added an index onto `users_id` the query was immediately improved. This is demonstrated here where I have highlighted the times that we see would see the issue comparing "yesterday" to "today":
![image](https://user-images.githubusercontent.com/3280530/220747579-7d3d0e3e-b7eb-43a9-a78e-66fa33db5e8f.png)
